lms/djangoapps/commerce/views.py

```diff
@@ -70,6 +71,15 @@ def _enroll(self, course_key, user):
         """ Enroll the user in the course. """
         add_enrollment(user.username, unicode(course_key))
 
+    def _get_honor_mode(self, course_key):
+        """ Retrieve the honor mode for the specified course. """
+        try:
+            honor_mode = CourseMode.objects.get(course_id=course_key, mode_slug=CourseMode.HONOR, currency="usd")
+        except ObjectDoesNotExist:
+            honor_mode = None
+
+        return honor_mode
```

Comment:
We tend to centralize logic like this in the `CourseMode` model. I'd much rather have it there rather than having this query the DB directly.